### PR TITLE
Add a Changelog section in README

### PR DESCRIPTION
**Pull request by xianzhidu**
Can be helpful for research project releases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,3 +36,7 @@ vision, and speech recognition and contains baseline configurations for training
 models.
 
 Please see [Concepts](docs/02-concepts.md) for more details on the core components and design of AXLearn, or [Getting Started](docs/01-start.md) if you want to get your hands dirty.
+
+## Changelog
+
+- [2024/06/28] Our Mixture-of-Experts (MoE) implementation and sharding strategy are released [[code]](https://github.com/apple/axlearn/blob/main/axlearn/common/mixture_of_experts.py). We also released a [tech report](https://arxiv.org/abs/2405.15052) that thoroughly studies the MoE LLMs and dense LLMs performance comparisons across a wide range of model scales.
```
Comment (markblee) [inline README.md:42]: A few comments:
1. Should we use a permalink for the code pointer?
2. Should we keep the `CHANGELOG.md` in its own file, and link to it from the ToC?

Comment (ruomingp) [reply README.md:42]: +1 on a separate file.

Also, it doesn't sound like a change log, but maybe a "models.md" file? We can add other entries such as Floris's sigmoid attention and Xiang's data filtering papers.

Comment (ruomingp) [reply README.md:42]: Also, in the future let's discuss the changes internally first.

Comment (xianzhidu) [reply README.md:42]: SG. Closing this one and discuss the comments internally first.

**Pull request closed without merging**